### PR TITLE
Docker Support

### PR DESCRIPTION
**Pull request by Auros**
This PR adds support for running Hive in a Docker container, along with some examples.

### CODE REVIEW

**Review by nike4613 — APPROVED** [review]
LGTM, also go ahead and add yourself as codeowner for the docker stuff.

**Review by nike4613 — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+COPY ["src/Hive.Plugins/Hive.Plugins.csproj", "src/Hive.Plugins/"]
+COPY ["src/Hive.Analyzers/Hive.Analyzers.csproj", "src/Hive.Analyzers/"]
+COPY ["src/CodeGen/Hive.CodeGen/Hive.CodeGen.csproj", "src/CodeGen/Hive.CodeGen/"]
+COPY ["src/CodeGen/Hive.CodeGen.Attributes/Hive.CodeGen.Attributes.csproj", "src/CodeGen/Hive.CodeGen.Attributes/"]
+COPY ["src/Hive.Utilities/Hive.Utilities.csproj", "src/Hive.Utilities/"]
+COPY ["src/Hive.Versioning/Hive.Versioning.csproj", "src/Hive.Versioning/"]
+COPY ["src/Hive.Permissions/Hive.Permissions.csproj", "src/Hive.Permissions/"]
+COPY ["src/Hive.Dependencies/Hive.Dependencies.fsproj", "src/Hive.Dependencies/"]
+RUN dotnet restore "src/Hive/Hive.csproj"
+COPY . .
```
Comment (sc2ad) [inline Dockerfile:21]: Does this not just copy everything in `.`? Did CWD change from `dotnet restore`?

**Review by sc2ad — APPROVED** [review]
One question, otherwise approvable

File: Dockerfile (hunk 1 above)
Comment (Auros) [reply Dockerfile:21]: I'm not entirely sure. However, [Microsoft recommends](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/docker/building-net-docker-images?view=aspnetcore-5.0#the-dockerfile) you restore the the project first and THEN copy everything else over for `Dockerfile`s.

**Pull request merged**